.github/workflows/ci_pipeline.yml

```diff
@@ -90,9 +90,9 @@ jobs:
         with:
           api_token: ${{ secrets.GITHUB_TOKEN }}
           update_packager_index: false
-          override_cache_key: ccache-${{ matrix.ros_distro }}-${{ github.head_ref || github.ref }}
+          override_cache_key: ccache-${{ matrix.ros_distro }}-${{ github.head_ref || github.ref_name }}
           override_cache_key_fallback: |
-            ccache-${{ matrix.ros_distro }}-${{ github.base_ref || github.ref }}
+            ccache-${{ matrix.ros_distro }}-${{ github.base_ref || github.ref_name }}
```

Comment:
nit: I've been thinking that we don't really benefit from splitting the caches from different branches, since it prevents main from benefiting from the final run from each PR, because the PR cache is lost. Splitting probably makes sense in repositories with lots of traffic, with multiple PRs running CI at the same time which would overwrite each other's cache upon merging.

ccache on its own can keep track of what is common and what is not. The one thing to keep under control is that in reusing the cache, it may never be purged and it will continue to grow, but we can cap the size to something sensible so that it begins evicting items if it grows to large. A full build is about 20-30Mb, so 300 Mb is probably a reasonable cap.

Reply:
Sounds like we could do that. The current behavior is not that bad though:
- A new PR can use main's cache, then it creates its own cache file and can re-use that after a new push.
- When the PR is merged, the first option is to use main's cache, which sounds sane IMO.

> The one thing to keep under control is that in reusing the cache, it may never be purged and it will continue to grow, but we can cap the size to something sensible so that it begins evicting items if it grows to large.

I'd have to see how this is implemented, but it seems like extra steps for only a small gain.

I think the main benefit of caching compiler output is to speed up builds before merging. The CI run after a push to main can usually wait without affecting any users.

Keeping cache sizes small is also good for upload/download times, although the difference between 30MB and 300MB can be marginal.

Reply:
Resolving the conversation to merge this PR, but we can follow-up on this and create a ticket if necessary.